client/client_update.go

```diff
@@ -235,10 +263,16 @@ func processUpdateResponse(response *http.Response) (interface{}, error) {
 		log.Debug("Have update available")
 		var ur UpdateResponse
 		if err = json.Unmarshal(respBody, &ur); err != nil {
-			return nil, errors.Wrap(err, "failed to parse the HTTP update response")
+			// In case the update control map was invalid, reparse
+			// without it to get the UpdateInfo. Doesn't matter if
+			// it fails, it will just be nil then, and we are
+			// already returning an error.
+			ur.UpdateControlMap.ignoreMap = true
+			_ = json.Unmarshal(respBody, &ur)
```

Comment:
Suggestion:
Simply pass in the `UpdateInfo` structure, and only that part will be unmarshalled

Reply:
Good idea!